### ISSUE TITLE
[CLOSED] How about move pytorch_transformer to sent_encoder level?

### ISSUE THREAD

**Issue by jeswan** [nyu-mll/jiant-v1-legacy#878]
<a href="https://github.com/HaokunLiu"><img src="https://avatars0.githubusercontent.com/u/35565210?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [HaokunLiu](https://github.com/HaokunLiu)**
_Monday Aug 05, 2019 at 01:56 GMT_
_Originally opened as https://github.com/nyu-mll/jiant/issues/878_

----

In practice, when ever using pretrained transformer-based models, we always use sent_encoder as "none", null phrase layer, and no highway layer. It's identical to using transformer as sent_encoder directly.

Plus, there have been researches showing BERT, unlike ELMo, does not benefit from more task-specific layers on top, and another paper showing this kind of model is more similar to a fusion of tradition pipeline, rather than simply a word-level embedding module. Therefore, in the near future, there probably won't be any need to pile layers on top of pretrained transformer models.

In addition, this simplify things a lot. If we consider pytorch_transformer module at the same level as sentence encoder, we can directly output the mask in our pytorch_transformer interface. And we no longer need to worry about the strange config terms unrelated to BERT, XLNet etc in sentence_encoder module, when writing config file for tranformer-based experiments.
